service/worker/migration/force_replication_workflow.go

```diff
@@ -207,15 +240,20 @@ func listWorkflowsForReplication(ctx workflow.Context, workflowExecutionsCh work
 
 		workflowExecutionsCh.Send(ctx, listResp.Executions)
 
-		params.NextPageToken = listResp.NextPageToken
+		params.Lock()
 		params.LastCloseTime = listResp.LastCloseTime
 		params.LastStartTime = listResp.LastStartTime
-
-		if params.NextPageToken == nil {
-			break
+		params.Unlock()
+
+		params.Token.NextPageToken = listResp.NextPageToken
```

Comment:
it would be more readable if we can wrap the logic here in a function like `params.Token = next( listResp.NextPageToken, params.Token)